### ISSUE TITLE
[ContactSensor] Opaque indexing of second tensor dimension N

### ISSUE THREAD

**Issue by bd-pdomanico** [mujocolab/mjlab#914]
The `ContactSensor` class has the following:

```
Output shape: [B, N * num_slots] or [B, N * num_slots, 3] where N = # of primaries
```

```
      n_primary = len(set(slot.primary_name for slot in self._slots))
      self._air_time_state = _AirTimeState(
        current_air_time=torch.zeros((n_envs, n_primary), device=device),
        last_air_time=torch.zeros((n_envs, n_primary), device=device),
        current_contact_time=torch.zeros((n_envs, n_primary), device=device),
        last_contact_time=torch.zeros((n_envs, n_primary), device=device),
        last_time=torch.zeros((n_envs,), device=device),
      )
```

which would imply that for air time state the second dim is `# of primary`, not multiplied by number of slots

## Recommendation

1. Update comments to make clear that dim N for air time is different than N for the other fields
2. Expose list of primary names (cache `_resolve_primary_names`) to facilitate reading per-primary data from the sensor

**OR**, is it the case that the design intent of a single contact sensor is not intended to be used for per-primary contact data and that several independent contact sensors should be used for such a use case (e.g. for locomotion: 1 contact sensor with a regex for all feet or n_legs contact sensors)

**Comment by kevinzakka** [comment]
Thanks for flagging this. You were right that the shape conventions were ambiguous, and digging in I found a real bug too: num_slots > 1 with track_air_time=True and more than one primary crashed in `_update_air_time_tracking`. #916 fixes that, clarifies that per-contact fields are `[B, P * num_slots, ...]` and per-primary air-time fields are `[B, P]`, and exposes `ContactSensor.primary_names` so you can map columns back to primaries. To your design question: a single sensor is fine for per-primary contact data; you don't need one sensor per foot.
